src/pages/DeviceSensor.tsx

```diff
@@ -0,0 +1,217 @@
+import { Box, Typography, Breadcrumbs } from "@mui/material";
+import RowContainerBetween from "../components/shared/RowContainerBetween";
+import { useLocation, useNavigate, useOutletContext, useParams } from "react-router-dom";
+import Chart from 'react-apexcharts';
+import { useCallback, useLayoutEffect, useState } from "react";
+import type { Actuator, Device, Sensor } from "waziup";
+import { Link } from "react-router-dom";
+import PrimaryIconButton from "../components/shared/PrimaryIconButton";
+import SensorTable from "../components/ui/DeviceTable";
+import { cleanString } from "../utils";
+function Device() {
+    function handleClick(event: React.MouseEvent<Element, MouseEvent>) {
+        event.preventDefault();
+    }
+    const { pathname } = useLocation();
+    const [device, setDevice] = useState<Device | null>(null);
+    const [sensOrActuator, setSensOrActuator] = useState<Sensor | Actuator | null>(null)
+    const [values, setValues] = useState<{ value: number | string, modified: string }[]>([]);
+    const [valsLimit, setValsLimit] = useState<number>(700);
+    const [graphValues, setGraphValues] = useState<{ y: number, x: string }[]>([]);
+    const [matches] = useOutletContext<[matches: boolean]>();
+
+    const navigate = useNavigate();
+    const { id, sensorId } = useParams();
+    const getGraphValues = useCallback(function (deviceId: string, sensorId: string) {
+        window.wazigate.getSensorValues(deviceId, sensorId)
+            .then((res) => {
+                const values = (res as { time: string, value: number }[]).map((value) => {
+                    const date = new Date(value.time);
+                    const hours = String(date.getUTCHours()).padStart(2, '0');
+
+                    const minutes = String(date.getUTCMinutes()).padStart(2, '0');
+                    return { 
+                        y: Math.round(value.value * 100) / 100,
+                        x: `${hours}:${minutes}`
+                    }
+                });
+                setGraphValues(values);
+                const valuesTable = (res as { time: string, value: number }[]).map((value) => {
+                    const date = new Date(value.time);
+                    const hours = String(date.getUTCHours()).padStart(2, '0');
+
+                    const minutes = String(date.getUTCMinutes()).padStart(2, '0');
+                    return {
+                        value: Math.round(value.value * 100) / 100,
+                        modified: `${date.getFullYear()}-${(date.getMonth()+1)}-${date.getDate()} ${hours}:${minutes}`
+                    }
+                })
+                setValues(valuesTable);
+            })
+    }, []);
+    async function fetchMoreData() {
+        let newValsx:{time:string,value: number}[]=[];
+        if(pathname.includes('actuators')){
```

Comment:
It's a bad idea to analyse the pathname in this way. There should be a better way? In any case, the "type of interface" should be resolved at a higher level in this file.